src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/waypoints_pending_resumption_handler.cc

```diff
@@ -139,71 +144,61 @@ void WayPointsPendingResumptionHandler::on_event(
     const application_manager::event_engine::Event& event) {
   using namespace application_manager;
   SDL_LOG_AUTO_TRACE();
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+
+  unsubscribe_from_event(Navigation_SubscribeWayPoints);
+
+  if (pending_requests_.empty()) {
+    SDL_LOG_DEBUG("Not waiting for any response");
+    return;
+  }
 
   const smart_objects::SmartObject& response = event.smart_object();
   const uint32_t corr_id = event.smart_object_correlation_id();
 
   SDL_LOG_TRACE("Received event with function id: "
                 << event.id() << " and correlation id: " << corr_id);
 
-  smart_objects::SmartObject pending_request;
-  if (pending_requests_.find(corr_id) == pending_requests_.end()) {
-    SDL_LOG_ERROR("corr id " << corr_id << " NOT found");
-    return;
-  }
-  pending_request = pending_requests_[corr_id];
-  pending_requests_.erase(corr_id);
-  if (app_ids_.empty()) {
-    SDL_LOG_ERROR("app_ids is empty");
-    return;
-  }
-  uint32_t app_id = app_ids_.front();
-  app_ids_.pop();
-  auto app = application_manager_.application(app_id);
+  auto current_pending = pending_requests_.front();
+  pending_requests_.pop_front();
+
+  auto app = application_manager_.application(current_pending.app_id_);
   if (!app) {
-    SDL_LOG_ERROR("Application not found " << app_id);
+    SDL_LOG_DEBUG("Application not found " << current_pending.app_id_);
```

Comment:
THis is warning or error. 
THis situation probably means that app was disconnected during resumption

Reply:
Done in fface93be6b07b30b9fe605959ace2d9b18b8f77